File: 1685-sum-of-absolute-differences-in-a-sorted-array/1685-sum-of-absolute-differences-in-a-sorted-array.cpp
auto init = []() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    return 0;
}();
#include <vector>
#include <numeric>

class Solution {
public:
    std::vector<int> getSumAbsoluteDifferences(std::vector<int>& nums) {
        int n = nums.size();
        
        // Using long long to prevent integer overflow during sum operations
        long long total_sum = 0;
        for (int num : nums) {
            total_sum += num;
        }

        long long left_sum = 0;
        std::vector<int> ans(n, 0);

        for (int i = 0; i < n; ++i) {
            long long right_sum = total_sum - left_sum - nums[i];

            long long left_diff = (long long)nums[i] * i - left_sum;
            long long right_diff = right_sum - (long long)nums[i] * (n - i - 1);

            ans[i] = left_diff + right_diff;
            left_sum += nums[i];
        }

        return ans;
    }
};

File: 1685-sum-of-absolute-differences-in-a-sorted-array/1685-sum-of-absolute-differences-in-a-sorted-array.py
class Solution(object):
    def getSumAbsoluteDifferences(self, nums):
        n = len(nums)
        ans = [0] * n
        pre = []
        k, s = 0, 0
        suf = []
        for i in range(n):
            k += nums[i]
            pre.append(k)
            s += nums[n - i - 1]
            suf.append(s)
        suf = suf[::-1] 

        for i in range(n):
            x = nums[i] * i - (pre[i - 1] if i > 0 else 0)
            y = (suf[i + 1] if i + 1 < n else 0) - nums[i] * (n - i - 1)
            ans[i] = x + y
        return ans
            



            


# class Solution(object):
#     def getSumAbsoluteDifferences(self, nums):
#         n = len(nums)
#         result = [0] * n
#         for i in range(n):
#             val = 0
#             for j in range(n):
#                 if i != j:
#                     val += abs(nums[i] - nums[j])
#             result[i] = val
#         return result
        